### PR TITLE
API version bump v35.0 - fix issues

### PR DESCRIPTION
**Pull request by mikeletux**
This PR aims to solve a little issue with v35.0 from VCD API.

In version v35.0 onwards, attribute `OrgVdc` from struct `OpenApiReference` is no longer supported from the API and `OwnerRef` must be used instead.

### CODE REVIEW

**Review by Didainius — CHANGES_REQUESTED** [review]
Looks good, but a a few notices:
* Missing Changelog entry.
* resource_vcd_external_network_v2.go has some conditionals which could be cleaned up as we don't need to support older versions now.
* resource_vcd_nsxt_network_dhcp.go
* resource_vcd_vapp_vm.go has a specific function for 10.1 `isNetworkRemovedInVcd101`. 

We could also clean up quite a few ineffective condition checks for tests as we transition to higher API version

```sh
grep -Rin "35.0" *.go | grep  test
datasource_vcd_certificate_library_test.go:27:  if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
datasource_vcd_certificate_library_test.go:28:          t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
datasource_vcd_nsxt_alb_importable_cloud_test.go:22:    if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
datasource_vcd_nsxt_alb_importable_cloud_test.go:23:            t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
datasource_vcd_nsxt_alb_settings_test.go:22:    if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
datasource_vcd_nsxt_alb_settings_test.go:23:            t.Skip(t.Name() + " requires at least API v35.0 (VCD 10.2+)")
datasource_vcd_vapp_network_test.go:28: const maxLeaseTime = 3500
resource_vcd_certificate_library_test.go:25:    if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_certificate_library_test.go:26:            t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_nsxt_alb_cloud_test.go:25: if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_cloud_test.go:26:         t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_nsxt_alb_controller_test.go:24:    if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_controller_test.go:25:            t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_nsxt_alb_edgegateway_service_engine_group_test.go:21:      if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_edgegateway_service_engine_group_test.go:22:              t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_nsxt_alb_edgegateway_service_engine_group_test.go:122:     if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_edgegateway_service_engine_group_test.go:123:             t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_nsxt_alb_pool_test.go:26:  if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_pool_test.go:27:          t.Skip(t.Name() + " requires at least API v35.0 (VCD 10.2+)")
resource_vcd_nsxt_alb_pool_test.go:740: if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_pool_test.go:741:         t.Skip(t.Name() + " requires at least API v35.0 (VCD 10.2+)")
resource_vcd_nsxt_alb_service_engine_group_test.go:25:  if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_service_engine_group_test.go:26:          t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_nsxt_alb_settings_test.go:24:      if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_settings_test.go:25:              t.Skip(t.Name() + " requires at least API v35.0 (VCD 10.2+)")
resource_vcd_nsxt_alb_virtual_service_test.go:25:       if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_virtual_service_test.go:26:               t.Skip(t.Name() + " requires at least API v35.0 (VCD 10.2+)")
resource_vcd_nsxt_alb_virtual_service_test.go:476:      if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_nsxt_alb_virtual_service_test.go:477:              t.Skip(t.Name() + " requires at least API v35.0 (VCD 10.2+)")
resource_vcd_vapp_network_test.go:54:           "defaultLeaseTimeForUpdate":   "3500",
resource_vcd_vapp_network_test.go:104:          "defaultLeaseTimeForUpdate":   "3500",
resource_vcd_vapp_vm_multinetwork_test.go:58:   if vcdClient.Client.APIVCDMaxVersionIs("= 35.0") {
resource_vcd_vdc_group_test.go:26:      if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_vdc_group_test.go:27:              t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
resource_vcd_vdc_group_test.go:83:      if vcdClient.Client.APIVCDMaxVersionIs("< 35.0") {
resource_vcd_vdc_group_test.go:84:              t.Skip(t.Name() + " requires at least API v35.0 (vCD 10.2+)")
```


```sh
grep -Rin "34.0" *.go | grep  test
datasource_vcd_network_isolated_v2_nsxt_test.go:20:     if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_network_isolated_v2_nsxt_test.go:21:             t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
datasource_vcd_network_routed_v2_nsxt_test.go:20:       if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_network_routed_v2_nsxt_test.go:21:               t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
datasource_vcd_nsxt_app_port_profile_test.go:21:        if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_app_port_profile_test.go:22:                t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
datasource_vcd_nsxt_app_port_profile_test.go:67:        if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_app_port_profile_test.go:68:                t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
datasource_vcd_nsxt_app_port_profile_test.go:105:       if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_app_port_profile_test.go:106:               t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
datasource_vcd_nsxt_edge_cluster_test.go:28:    if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_edge_cluster_test.go:29:            t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1+)")
datasource_vcd_nsxt_edgegateway_test.go:24:     if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_edgegateway_test.go:25:             t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1+)")
datasource_vcd_nsxt_edgegateway_test.go:266:    if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_edgegateway_test.go:267:            t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1+)")
datasource_vcd_nsxt_network_imported_test.go:20:        if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
datasource_vcd_nsxt_network_imported_test.go:21:                t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
resource_vcd_network_isolated_v2_nsxt_test.go:20:       if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
resource_vcd_network_isolated_v2_nsxt_test.go:21:               t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
resource_vcd_network_routed_v2_nsxt_test.go:20: if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
resource_vcd_network_routed_v2_nsxt_test.go:21:         t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
resource_vcd_nsxt_app_port_profile_test.go:20:  if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
resource_vcd_nsxt_app_port_profile_test.go:21:          t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
resource_vcd_nsxt_app_port_profile_test.go:143: if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
resource_vcd_nsxt_app_port_profile_test.go:144:         t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
resource_vcd_nsxt_edgegateway_test.go:28:       if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
resource_vcd_nsxt_edgegateway_test.go:29:               t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1+)")
resource_vcd_nsxt_network_imported_test.go:20:  if vcdClient.Client.APIVCDMaxVersionIs("< 34.0") {
resource_vcd_nsxt_network_imported_test.go:21:          t.Skip(t.Name() + " requires at least API v34.0 (vCD 10.1.1+)")
resource_vcd_vapp_vm_hot_updates_test.go:82:    if vcdClient.Client.APIVCDMaxVersionIs("= 34.0") {
```

File: .changes/v3.6.0/795-improvements.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Update codebase to be compatible with VCD v35.0 [GH-795]
```
Comment (lvirbalas) [inline .changes/v3.6.0/795-improvements.md:1]: Let's provide a bit more context:

```suggestion
* Update codebase to be compatible with changes in go-vcloud-director due to bump to VCD API V35.0 [GH-795]
```

Please also add notes to PR description/merge commit message about changes in schema.Schema and diag usage.

**Review by lvirbalas — APPROVED** [review]
Small asks, LGTM otherwise!

File: .changes/v3.6.0/795-improvements.md (hunk 1 above)
Comment (mikeletux) [reply .changes/v3.6.0/795-improvements.md]: Changes done!
Also, I'll write in the merge commit the changes you suggest. Thanks!

**Review by adambarreiro — APPROVED** [review]
Nice cleanup!

**Comment by mikeletux** [comment]
@Didainius already done the changes you suggested. Please check 😄

**Review by Didainius — APPROVED** [review]

**Pull request merged**